File: client/src/app/modules/miscellaneous/services/civility.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { Civility } from '../../miscellaneous/model/Civility';

@Injectable({
  providedIn: 'root'
})
export class CivilityService extends AppRestService<Civility>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getCivilities() {
    return this.getListCached(new HttpParams(), "civilities");
  }

  addOrUpdateCivility(civility: Civility) {
    this.clearListCache(new HttpParams(), "civilities");
    return this.addOrUpdate(new HttpParams(), "civility", civility, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/CivilityService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.Civility;

public interface CivilityService {
    public List<Civility> getCivilities();

    public Civility getCivility(Integer id);

    public Civility addOrUpdateCivility(Civility civility);
}
